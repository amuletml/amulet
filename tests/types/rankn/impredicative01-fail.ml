(* This used to be a very simple test regarding polytypes under constructors,
 * but we allow that *)

type id 'a = Id of 'a

let foo : id (forall 'a. 'a -> 'a) = Id (fun x -> x)
