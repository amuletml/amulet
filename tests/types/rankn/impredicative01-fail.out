id : type -> type
Id : Spec{'a : type}. 'a -> id 'a
foo : id (Spec{'a : type}. 'a -> 'a)
